### PR TITLE
Premium Traitor Items [DNM]

### PR DESCRIPTION
**Pull request by ghost**
Simply put, this PR adds sections to the traitor uplink that contain special premium items. Certain jobs can buy these items at 50% of their normal price. For instance, a flask of powerful moonshine can be purchased by the Bartender for about 2 telecrystals, but an assistant can also buy it for 4.

This is largely incomplete (although functional) so please do not merge this.

### CODE REVIEW

**Comment by WJohn** [comment]
I swear this is on some big bad "do not attempt this feature" list like player references and copyright material.

Not that I ever found out why, so :+1:

**Comment by Iamgoofball** [comment]
this encourages playing as something other then assistant

:+1:

**Comment by dannno** [comment]
finally :+1:

**Comment by SconesC** [comment]
WEEOOOOO WEOOOOO PROBLEM ALERT

BARTENDER AND CHEM CANNOT BUY TOXINS KIT


WEOOOO WEOOOO

**Comment by Xhuis** [comment]
Chemist should work fine. I'll add bartender to the list though.

**Comment by Iamgoofball** [comment]
Actually yeah, I have a problem with that as the guy who added it.

Leave the toxins kit as a global item.

**Comment by dannno** [comment]
>as the guy who added it

I don't think this hold a lot of weight in any matter when it comes to rebalancing things.

**Comment by Incoming5643** [comment]
If you ask me if we have this at all it should only be applied to stupid gimmicky shit

File: code/datums/uplink_item.dm [hunk 1]
```diff
@@ -697,3 +690,71 @@ var/list/uplink_items = list()
...
+	item = /obj/item/weapon/storage/box/syndie_kit/chemical
+	cost = 6
+	jobs = list("Chemist")
+
+/datum/uplink_item/job_medsci/what_have_i_done
+	name = "Chlorine Trifluoride Foam Grenade"
+	desc = "A terrifying grenade that causes a gigantic area to be blanketed in chlorine trifluoride, designed to exterminate life. It is highly recommended you flee the scene before it goes off. \
+	Jobs: Chemist"
+	item = /obj/item/weapon/grenade/chem_grenade/clf3
+	cost = 11 //No way someone's getting more than one of these monsters
```
Comment (Incoming5643) [inline code/datums/uplink_item.dm]: it's out of 20, so either this remark was tongue in cheek or all the items you added are half the price you intended

**Comment by Niknakflak** [comment]
I ask that you make it so you can't see the other traitor items unless you are that job.  For example, instead of just one huge list where you can only buy this job specific traitor item but all are there for listing, make it so only the ones you can purchase based on job, show up.

File: code/game/objects/items/weapons/storage/boxes.dm [hunk 2]
```diff
@@ -122,6 +121,21 @@
...
+/obj/item/weapon/storage/box/lethalsyringes/New()
+	..()
+	new /obj/item/weapon/reagent_containers/syringe/lethal(src)
+	new /obj/item/weapon/reagent_containers/syringe/lethal(src)
+	new /obj/item/weapon/reagent_containers/syringe/lethal(src)
+	new /obj/item/weapon/reagent_containers/syringe/lethal(src)
+	new /obj/item/weapon/reagent_containers/syringe/lethal(src)
+	new /obj/item/weapon/reagent_containers/syringe/lethal(src)
+	new /obj/item/weapon/reagent_containers/syringe/lethal(src)
+
```
Comment (Incoming5643) [inline code/game/objects/items/weapons/storage/boxes.dm]: Unless I'm mistaken you can't actually fill lethal syringes once they're empty, so giving someone a box of empty ones is useless

**Comment by Xhuis** [comment]
@Niknakflak I couldn't figure out how to do that but I'll see what I can do.

@Incoming5643 The remark was intended. Clf3 grenades are horrifying. Also, you can fill up lethal syringes since they are just syringe subtypes.

**Comment by Steelpoint** [comment]
The title for this PR should be called "Chemist-Specific Traitor Items" as it seems the Chemist gets most of the fun.

**Comment by dannno** [comment]
pretty sure this is wip @Steelpoint ...............................................

**Comment by PKPenguin321** [comment]
finally, librarian traitors might be able to get something other than murdered
soul stones for chaplain when

**Comment by Tokiko1** [comment]
I think that cooks and bartenders should also have access to the poison kit. For spiking food and drinks. Possibly botanists aswell.

File: code/datums/uplink_item.dm [hunk 3]
```diff
@@ -697,3 +690,71 @@ var/list/uplink_items = list()
...
+	item = /obj/item/device/rad_laser
+	cost = 6
+	jobs = list("Medical Doctor", "Chemist", "Geneticist", "Virologist", "Chief Medical Officer")
+
+/datum/uplink_item/job_medsci/traitor_chem_bottle
+	name = "Poison Kit"
+	desc = "An assortment of deadly chemical solutions. Comes with a large variety, each fitting into their own niche. Jobs: Chemist"
+	item = /obj/item/weapon/storage/box/syndie_kit/chemical
+	cost = 6
+	jobs = list("Chemist")
```
Comment (Dorsisdwarf) [inline code/datums/uplink_item.dm]: Poison kit is 2versatile for this.

File: code/datums/uplink_item.dm [hunk 4]
```diff
@@ -697,3 +690,71 @@ var/list/uplink_items = list()
...
+
+/datum/uplink_item/job_medsci/what_have_i_done
+	name = "Chlorine Trifluoride Foam Grenade"
+	desc = "A terrifying grenade that causes a gigantic area to be blanketed in chlorine trifluoride, designed to exterminate life. It is highly recommended you flee the scene before it goes off. \
+	Jobs: Chemist"
+	item = /obj/item/weapon/grenade/chem_grenade/clf3
+	cost = 11 //No way someone's getting more than one of these monsters
+	jobs = list("Chemist")
```
Comment (Dorsisdwarf) [inline code/datums/uplink_item.dm]: >11 TC for something that can be easily and almost safely mass-produced by the only job it's available for

File: code/datums/uplink_item.dm [hunk 5]
```diff
@@ -697,3 +690,71 @@ var/list/uplink_items = list()
...
+	item = /obj/item/weapon/storage/box/lethalsyringes
+	cost = 8
+	jobs = list("Chemist")
+
+/datum/uplink_item/job_medsci/syringe_gun
+	name = "Syringe Gun"
+	desc = "We literally just pilfered this from another station. If someone grabs the other ones, then you can have this one. Jobs: Chemist"
+	item = /obj/item/weapon/gun/syringe
+	cost = 1
+	jobs = list("Chemist")
```
Comment (Dorsisdwarf) [inline code/datums/uplink_item.dm]: At least make it the rapidsyringe, this is pointless

**Comment by Dorsisdwarf** [comment]
Suggestion: Box of Airhorns for the clown. 1tc. It's a box full of airhorns.

**Comment by Miauw62** [comment]
No.
There will never be job-specific traitor items.
Items that are only useful for a few jobs are OK, but it is dumb to restrict items to certain jobs, this only limits creativity.

**Comment by Miauw62** [comment]
Sigh. going to re-open this and re-emphasize the DNM here.
I REALLY REALLY don't like this, but I guess it should be discussed.
The job-specific traitor items we have now are the things you can get by playing your job. A cargo tech traitor won't  be running around with ubervirusses, a chef traitor won't be unleashing a horde of slimes on the station, a clown won't flood the station with plasma, etc.

**Comment by Miauw62** [comment]
inb4 "april fools"

**Comment by Fox-McCloud** [comment]
I presume this is April Fools...if it isn't, big big :+1: from me

I don't see how these limit creativity so much as offer compensation for playing a particular job and rolling traitor when that job is otherwise fairly useless as traitor...Additionally, it's a bit silly to emphasize that "job specific traitor items" as being ok when they're a function of the map as opposed to a function of an uplink.

Giving the chef a special cleaver that's roughly the same price as the esword and has a similar force rating, but has some other special function to it isn't "limiting creativity", it's offering incentive to play that job--no one is forcing the chef to go with those items....and let's be honest, the chef (and a plethora of other jobs) have zero on most others---a scientist, engi, or medical traitor is almost always preferably to a low access traitor with hardly any unique gear.

**Comment by AlexCrimson** [comment]
Pretty sure there are plenty of forum topics full of job-specific traitor items.

Id suggest not limiting them to certain jobs. Allow any traitor to buy them. Maybe increase the TC cost for the item if its a different job class than your own? Traitors would then be able to frame others.

**Comment by Cheridan** [comment]
Other jobs being able to get job-specific items at a premium cost sounds like a pretty good idea, it solves the Miauw's issues of job changes/limiting content but still allows jobs to have interesting thematic items.

File: code/datums/uplink_item.dm (hunk 4 above)
Comment (Xhuis) [inline code/datums/uplink_item.dm]: Smoke no longer applies touch effects. In addition, this clf3 grenade has 290 units of fluorosurfactant, which is enough to fill about two thunderdomes.

**Comment by Xhuis** [comment]
I don't see why that wouldn't work.

**Comment by Iamgoofball** [comment]
I like Cheridan's idea. That solves the problem of limiting content.

**Comment by ACCount12** [comment]
:-1: Limiting items is bad. What if i'm botanist and i want to posion some apples? Or chef, or barman, or random assistant throwing antitox first aids with toxins instead of antitoxins in syringes.

**Comment by Xhuis** [comment]
@ACCount12 I'm not gonna limit the items. I'll update the PR with that. Instead, some jobs get discounts on items if it makes more sense for them to use. If a doctor wants to use a flask of moonshine, they can, it'll just cost them the normal amount, whereas the bartender gets a 50% discount.

**Comment by Fox-McCloud** [comment]
@ACCount12 

That same argument could be made for job based items that are non-traitor related. What If I want to be a botanist who releases slimes---or a chef who has a recharging  laser gun at round start? Balance--job specific traitor items accomplish the same purpose around a common theme; they give a small boost to a job that otherwise doesn't have a great boost or offer a skill-based item that allows people who are already normally good at their jobs to be rewarded for that in a traitorous way.

**Comment by ACCount12** [comment]
@Fox-McCloud, no, it's not related. Traitoring is above all jobs. You are just limiting traitors creativity. As traitor clown, i can release toxins or slimes (framing atmos or sci) with nothing but toolbox and a bit of creativity, but i can't buy a medbay-only item. It's just pointless.

There is no point in Syndicate limiting some items it has to jobs too. For Syndicate, any agent is agent, no matter is he assistant or sciencist. 

And limiting items to jobs encourages meta.

**Comment by Dorsisdwarf** [comment]
Please note that items are no longer limited by-job. Instead, certain jobs get a big discount on certain items.

**Comment by AlexCrimson** [comment]
Not really related to the PR, but id like to take this moment to remind people that i also suggested something similar a while ago...

Split up all traitor items into groups. Stealth/Lethal/Sabotage/Whatever. Then depending on your objectives you would get a discount or need to pay more for certain items. Stealthy objectives would get cheap stealth items but expensive lethal items. Murder objectives would get cheap lethal items but expensive sabotage items etc.

**Comment by Kelenius** [comment]
>Split up all traitor items into groups. Stealth/Lethal/Sabotage/Whatever. Then depending on your objectives you would get a discount or need to pay more for certain items. Stealthy objectives would get cheap stealth items but expensive lethal items. Murder objectives would get cheap lethal items but expensive sabotage items etc.

Terrible idea.

**Comment by AlexCrimson** [comment]
and discourages murderboning when your objective is to just steal something.

**Comment by Miauw62** [comment]
and discourages people from doing anything but their objectives. yes,
that'll certainly make the game more interesting.

**Comment by babinsquared** [comment]
I like it when people get creative and would be sad to see chefs/botanists/etc be unable to get chem poisons, and so on. Part of the fun of the game is figuring out interesting ways to accomplish objectives and this just kind of limits them.

**Comment by WJohn** [comment]
I think if you're going to make a discount, you should only create discounts, not raise prices on anything. It'd encourage greentexting but it wouldn't force your hand when you originally could do it. Yes it'd be a straight buff to traitors but honestly they're kind of wimpy in the face of all the new curve balls we're throwing at them left and right.

**Comment by Xhuis** [comment]
The thing is, it can either be viewed as price-raising or discounting. I could have the base cost as 3 but it costs 6 for people not in that job, or I can have the base price 6 and it costs 3 for people in the job. I'm not going to create anything that favors one job over another, and I will do balancing to prevent that.

**Comment by AlexanderUlanH** [comment]
tl;dr, a player's innate job is a hard-coded variable that can't be interacted with in gameplay, so it shouldn't be used for something as important as traitor item selection.

Job-specific things are set in stone forever at roundstart, meaning that they can't be changed ever through gameplay.  A cargo tech can get some illegal things by using an emag on a cargo shuttle.  It's difficult for anyone but a cargo tech to do the same, because they don't have the ID access to the request computers, or the right to be in the cargo department.  But those are all things that can be overcome ingame, and it's pretty interesting when they are. (ie disguising as a cargo tech, buddying up with / bribing them cargo techs, transferring departments, etc)
If, instead, people who were cargo techs at roundstart had an innate ability to access syndie gear via the consoles, that would remove all of those potential situations.
The bartender doesn't need a discount on poisons, because he's in a position where it would be more useful to him in the first place, a position which someone who's not a bartender might find be able to maneuver himself into.

I like the idea of certain items typically being used by traitors of certain jobs, but I think that should be accomplished by something more in-character and involved than "the syndicate wants you to do something thematically appropriate for the job you signed up as so here's a discount."

**Comment by pudl** [comment]
put id into pda > get the discounts of the job on the id

**Comment by Xhuis** [comment]
I'm still having trouble with this, so I'm gonna close it. Anyone who wants to continue it can feel free to copy the code.
